craft_providers/lxd/installer.py

```diff
@@ -89,3 +98,41 @@ def is_installed() -> bool:
     :returns: True if lxd is installed.
     """
     return shutil.which("lxd") is not None
+
+
+def is_user_permitted() -> bool:
+    """Check if user has permisisons to connect to LXD.
+
+    The user must be root or be in lxd group to talk to the local LXD socket.
+
+    :returns: True if user has correct permissions.
+    """
+    return (
+        any(grp.getgrgid(g).gr_name == "lxd" for g in os.getgroups())
+        or os.geteuid() == 0
```

Comment:
Probably better to test for euid first so you don't have to check groups if user is root?

Reply:
fair point!

Reply:
done!